server/controllers/googleApiController.js

```diff
@@ -82,38 +89,38 @@ function authenticateUser(req, res) {
     .catch((err) => res.status(422).json(err));
 }
 
-function getAvailability(req, res) {
+async function getAvailability(req, res) {
   const { query } = req;
   const date = `${query.year}/${query.month}/${query.date}`; // 2020-08-20
   const meetingLength = parseInt(query.meetingLength); // "30min" => 30
-  const { availableFrom, availableTo, timeZone } = query;
-
-  //create time availability range based on user availability preference
-  const availabilityStart = moment(`${date} ${availableFrom}`);
-  const availabilityEnd = moment(`${date} ${availableTo}`);
-  let timeSlot = createTimeSlot(
-    availabilityStart,
-    availabilityEnd,
-    meetingLength
-  );
+  const { timeZone, calendarUrl } = query;
+  let user;
 
-  //TO DO: ================================================
-  //get refresh_token from database
-  //=======================================================
-  let tokens = {
-    refresh_token:
-      "1//04_ZIKYvcn2LOCgYIARAAGAQSNwF-L9Ir-x_TcEaoH93ZbHi-BFl334mhzcedu6jsgXCMCAUH_H8CqF_IIIfRXiR32v7tiCt2OvM",
-  };
+  findUserByUrl(calendarUrl)
+    .then((dbModel) => {
+      user = dbModel;
+      user.availabilityStart = moment(`${date} ${user.availableHoursFrom}`);
```

Comment:
hm how come this date it stored as a model attribution? i'm guessing this is not meant to be stored in the db. 

if true, you can either create a method on the model or similar to before, keep the variable

Reply:
it is not saved into the database.I'm using `.then` chain, and the **availabilityStart /End** are used in multiple` .then()` function but not passed down as a response.  

The easiest way I can think of is saving them as global variables outside the then chain, like **user** in line 97, but I don't want too many global variables. Since **availabilitystart/end** are part of **user** info, so I saved them into user object as a property. 

**User** is a local variable to the getAvailability function, will not be saved to DB, nor sent to the client. so I think it would be fine adding properties.

Reply:
changed them back to global variable

Reply:
👍  changing these into variables are not really "global" because they're still within the scope of this function. since availabilityStart and availabilityEnd is very specific to this particular `date`, i would either keep it as a variable or create a method in the User model to return this `availabilityStart` value